api/app/controllers/spree/api/v1/shipments_controller.rb

```diff
@@ -85,11 +85,22 @@ def transfer_to_location
         def transfer_to_shipment
           @target_shipment = Spree::Shipment.find_by!(number: params[:target_shipment_number])
 
-          if @quantity < 0 || @target_shipment == @original_shipment
-            unprocessable_entity('ArgumentError')
+          message = 'Following errors occured while attempting this action:'
+
+          if @quantity < 0 && @target_shipment == @original_shipment
+            message += " #{Spree.t('negative_quantity', scope: 'api').downcase}, #{Spree.t('wrong_shipment_target', scope: 'api').downcase}"
+            unprocessable_entity(message)
+            return
+          elsif @target_shipment == @original_shipment
+            message += " #{Spree.t('wrong_shipment_target', scope: 'api').downcase}"
+            unprocessable_entity(message)
```

Comment:
If you're just appending to message variable, you can call `unprocessable_entity(message)` once outside of if/elsif block